### DISCUSSION TITLE
Foundational documents for the community

### DISCUSSION THREAD

**Discussion by raskyld** [wavetermdev/waveterm#84] [Ideas]
Hello folks,

Awesome initiative. 👏 !

I am interested in a few things:

- 👀 The vision of the maintainers on _wave_,
- 🚀 An _overview_ roadmap (maybe with per-Quarter or per-Version features?),
- 📐 An architecture overview of the project.

## Vision 👀 

Would it be possible from maintainers to formalise their vision somewhere? (Maybe it's already done and I just didn't RTFM well enough 🙈), as contributors, we are more effective when we share the same vision than core maintainers as it allows us to:

1. Have an _almost_ identical triage process of issues and features, we know what is important for the project and what is not.
2. We feel bound by core values, which bring the community closer together. 👐 

I am a Platform Engineer working for an 🇪🇺 CSP, it's kind of in my DNA to make Developer Tools more inclusive. I am pretty sure I can also raise awareness in the future within my company (@scaleway) on the _wave_ project and help integrating our CLI tools in _wave_ once there is extensibility mechanisms.

I am also deeply interested in developing **Kubernetes** integration in _wave_: Kubernetes is a tool most developers will meet at some point in their career 🌱. Making Kubernetes more user-friendly through terminal integration (observability, auto-completion) would be awesome and in-line with what _I think is_ the philosophy of wave.

## Roadmap 🚀 

I saw a lot of discussions about the roadmap of Wave.
Would it be possible to write a document that explains where we want to focus the strength of the community?
Or maybe using GitHub milestones would be better?

## Architecture

From early exploration, it seems like Wave is an electron-based app with a server written in Go, can we have more details on the tools used and/or an overview of the architecture? 🙏 

Thanks a lot!

**Comment by sawka** [comment]
Thanks for posting this.  These are great ideas.  I'm onboard for posting a quarterly roadmap document to give more insight into our internal process.  Also definitely owe the community an architecture diagram -- it's been a good learning experience getting electron + Go working together (I am excited about the combo).

Also FYI, we're adding a couple of new full-time engineers in the next few weeks which will give us a lot more internal capacity.

**Comment by sawka** [comment]
I'd like to break out the k8s integration into a separate discussion.  I think Wave can really improve the DevEx of kubernetes if we design it right and would love to collect ideas around what would be most useful.
